src/core_plugins/i18n/server/i18n/i18n.js

```diff
@@ -0,0 +1,124 @@
+import path from 'path';
+import Promise from 'bluebird';
+import { readFile } from 'fs';
+import _ from 'lodash';
+
+const asyncReadFile = Promise.promisify(readFile);
+
+const TRANSLATION_FILE_EXTENSION = '.json';
+
+let defaultLocale = 'en';
```

Comment:
Keeping module state like this isn't going to work if multiple Kibana servers are started in the same process with different configurations (A multi-tenancy strategy, and often done in the tests). Instead state should be tracked in something that is server specific (an instance of a class, or just a closure around an object).

If you like I think it would make sense to extend the [`UiI18n` class from my pull request to your fork](https://github.com/spalger/kibana/blob/3122d94b3e4f3ebfc0ce43c7e67c8a88d815250c/src/ui/ui_i18n.js) with some of the methods necessary to do the `_build:verifyTranslations` grunt task, but I'll leave that up to you.

Moving this scope into an object that is server specific should also simplify the tests some, as they won't need to constantly reset the state of the i18n module.